### DISCUSSION TITLE
assetpack-plugin-audiosprite – Pack hundreds of sounds into one compressed download

### DISCUSSION THREAD

**Discussion by reececomo** [pixijs/assetpack#68] [Show and tell]
Have published a little wrapper plugin for `audiosprite`: [assetpack-plugin-audiosprite](https://www.npmjs.com/package/assetpack-plugin-audiosprite)

```
npm install assetpack-plugin-audiosprite --save-dev
```

### Basic usage

```js
// .assetpack.js
const { audiosprite } = require('assetpack-plugin-audiosprite');

module.exports = {
  audiosprite: audiosprite()
};
```

> [!NOTE]
> [See README.md for plugin options here](https://github.com/reececomo/assetpack-plugin-audiosprite#options)

<details><summary><h3>Example usage</h3></summary>

#### Input files

```
raw-assets/
    my_sounds{sfx}/
        cry.m4a
        laugh.wav
        sneeze.mp3
        whine.mp3
        yell.aiff
```

#### .assetpack.js config

```js
  audiosprite: audiosprite({
    tags: { audiosprite: 'sfx' }, // default: 'audiosprite'
    audiosprite: {
      // configure 'audiosprite' options:
      export: 'mp3,ogg',
      bitrate: 64,
      samplerate: 32_000
    }
  }),
```

#### Output:
```
assets/
    my_sounds/
        my_sounds.json
        my_sounds.mp3
        my_sounds.ogg
```

</details>

**Comment by reececomo** [comment]
Here's a quick extension to load the files [audiospriteAsset.ts](https://gist.github.com/reececomo/c7d7c34589dd2473049608789811512d) (pixi v7, but should be same/pretty close to v8)

### Usage

```ts
// register loader
extensions.add(audiospriteAsset);
```

```ts
// load sprites
const mySounds = await Assets.load('assets/sfx.json');
mySounds.play('bark');
```

**Comment by Zyie** [comment]
hey @reececomo this is a great plugin. We just released 1.0 of assetpack which pretty significantly changes the API for plugins. If you have any questions/issues with updating the plugin to work with the latest release let me know and i will try to help

**Reply by reececomo** [reply]
Thanks @Zyie — looks good! This audiosprite plugin is small, would you have thoughts on a TS version as PR to https://github.com/pixijs/assetpack?

**Reply by Zyie** [reply]
yeah if you want to make a PR I'll add this plugin to assetpack!

**Reply by furic** [reply]
Any update @reececomo?
I actually created [an issue](https://github.com/pixijs/assetpack/issues/83) before seeing this, and mentioned my expected usage.

**Reply by reececomo** [reply]
@furic we're still currently using [the version above](https://www.npmjs.com/package/assetpack-plugin-audiosprite), as we're on the previous release of assetpack.

If you look at the source code its probably fairly easy to adapt!

If you give [this code](https://github.com/reececomo/assetpack-plugin-audiosprite/blob/main/plugin.js) to ChatGPT/Claude I'm sure it could probably help generate you an AssetPack v1.0 TS plugin 😄
